core/src/commonMain/kotlin/io/github/aag/core/data/network/sources/CoursesRemoteDataSource.kt

```diff
@@ -0,0 +1,29 @@
+package io.github.aag.core.data.network.sources
+
+import io.github.aag.core.data.network.ktor.CoursesResource
+import io.github.aag.core.data.network.models.CourseDTO
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.resources.get
+import io.ktor.client.utils.clientDispatcher
+import io.ktor.util.InternalAPI
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+private const val DEFAULT_THREAD_COUNT = 64
+
+@OptIn(InternalAPI::class)
+class CoursesRemoteDataSource(
+    private val httpClient: HttpClient
+) {
+    suspend fun getAllCourses(): List<CourseDTO> =
+        withContext(
```

Comment:
delegate it to platform-specific HttpEngine

Reply:
As I see [here](https://github.com/ktorio/ktor/blob/main/ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpEngine.kt), okhttp engine uses its own coroutine context under the hood but I don't understand how thread pool switch happens. As I saw during debug (coming down to `io.ktor.client.HttpClient#execute` -> `io.ktor.util.pipeline.SuspendFunctionGun#execute`) there still was a main thread selected as active.

Reply:
Actually, my context switching here also doesn't help - it's nearly the same as okhttp one, so I removed it. Though, I don't see any warning like "network request from main thread" 🤔  (while I was debugging and checking commiunnication with back-end)